contracts/SaveDAI.sol

```diff
@@ -13,14 +13,17 @@ import "./lib/CTokenInterface.sol";
 import "./lib/OTokenInterface.sol";
 import "./lib/ISaveDAI.sol";
 
-contract SaveDAI is ISaveDAI, ERC20, ERC20Detailed, ERC20Pausable, Ownable {
+contract SaveDAI is ISaveDAI, ERC20, ERC20Detailed, Pausable, Ownable {
     using SafeMath for uint256;
 
     /***************
     GLOBAL CONSTANTS
     ***************/
     // Variable to set distant deadline for Uniswap tokenToTokenSwap transactions
-    uint256 constant LARGE_BLOCK_SIZE = 1651753129000;
+    uint256 constant LARGE_BLOCK_SIZE = 1099511627776;
+
+    // Variable used to set near infinite approval allowances
+    uint256 constant LARGE_APPROVAL_NUMBER = 10**30;
```

Comment:
10^12  tokens isn't really THAT large. A dedicated attacker could flash loan, say 100m DAI (currently not available, but might be in the future?), and move it in and out of the system (so long as there isn't a fee or something that reduces their owned value), each time reducing the tracked approval. It would take ~10K tx, but not impossible.

If you make it the max int (uint256(-1)), it avoids this issue AND saves gas, due to this code in DAI:

```
        if (src != msg.sender && allowance[src][msg.sender] != uint(-1)) {
            require(allowance[src][msg.sender] >= wad, "Dai/insufficient-allowance");
            allowance[src][msg.sender] = sub(allowance[src][msg.sender], wad);
        }
```

Skips tracking allowance for that one specific max int value.

Reply:
Interesting, ty!